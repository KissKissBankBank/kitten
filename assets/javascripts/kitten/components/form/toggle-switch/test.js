import React from 'react'
import renderer from 'react-test-renderer'
import { ToggleSwitch } from './index'

describe('<ToggleSwitch />', () => {
  let component

  describe('with default props', () => {
    beforeEach(() => {
      component = renderer.create(<ToggleSwitch id="toggle-switch" />).toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `label` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ToggleSwitch id="toggle-switch" label="A test label" />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `isChecked` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ToggleSwitch id="toggle-switch" isChecked />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `big` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ToggleSwitch id="toggle-switch" big />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `disabled` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ToggleSwitch id="toggle-switch" disabled />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `isLabelVisible` prop', () => {
    beforeEach(() => {
      component = renderer
        .create(<ToggleSwitch id="toggle-switch" isLabelVisible={false} />)
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })

  describe('with `checkedColor` `defaultColor` & `disabledColor` props', () => {
    beforeEach(() => {
      component = renderer
        .create(
          <ToggleSwitch
            id="toggle-switch"
            checkedColor="#000"
            defaultColor="#000"
            disabledColor="#000"
          />,
        )
        .toJSON()
    })

    it('matches with snapshot', () => {
      expect(component).toMatchSnapshot()
    })
  })
})
