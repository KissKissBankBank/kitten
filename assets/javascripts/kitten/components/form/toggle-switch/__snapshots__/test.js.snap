// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleSwitch /> with \`big\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH jyMTyt"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat dNJSGI"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;

exports[`<ToggleSwitch /> with \`checkedColor\` \`defaultColor\` & \`disabledColor\` props matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH dfHyXA"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat hadUAa"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;

exports[`<ToggleSwitch /> with \`disabled\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa cXgxJA"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH jyMTyt"
    disabled={true}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat dbfzSp"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;

exports[`<ToggleSwitch /> with \`isChecked\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={true}
    className="sc-bwzfXH jyMTyt"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat dbfzSp"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;

exports[`<ToggleSwitch /> with \`isLabelVisible\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label="switch"
    aria-labelledby={null}
    aria-pressed={false}
    className="sc-bwzfXH jyMTyt"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
</div>
`;

exports[`<ToggleSwitch /> with \`label\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH jyMTyt"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat dbfzSp"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    A test label
  </label>
</div>
`;

exports[`<ToggleSwitch /> with \`locked\` prop matches with snapshot 1`] = `
<div
  className="sc-bdVaJa cXgxJA"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH UUQdg"
    disabled={true}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    >
      <svg
        fill="var(--color-grey-900, #222)"
        height={15.42857142857143}
        viewBox="0 0 14 18"
        width="12"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5,6 L9,6 L9,4 C9,2.8954305 8.1045695,2 7,2 C5.8954305,2 5,2.8954305 5,4 L5,6 Z M11,6 L14,6 L14,18 L0,18 L0,6 L3,6 L3,4 C3,1.790861 4.790861,0 7,0 C9.209139,0 11,1.790861 11,4 L11,6 Z M6.5,11.9146471 L6.5,14.5 C6.5,14.7761424 6.72385763,15 7,15 C7.27614237,15 7.5,14.7761424 7.5,14.5 L7.5,11.9146471 C8.08259619,11.7087289 8.5,11.1531094 8.5,10.5 C8.5,9.67157288 7.82842712,9 7,9 C6.17157288,9 5.5,9.67157288 5.5,10.5 C5.5,11.1531094 5.91740381,11.7087289 6.5,11.9146471 Z"
        />
      </svg>
    </div>
  </button>
  <label
    className="sc-htpNat dbfzSp"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;

exports[`<ToggleSwitch /> with default props matches with snapshot 1`] = `
<div
  className="sc-bdVaJa dPVrAh"
>
  <button
    aria-label={null}
    aria-labelledby="toggle-switch_label"
    aria-pressed={false}
    className="sc-bwzfXH jyMTyt"
    disabled={false}
    id="toggle-switch"
    onClick={[Function]}
    type="button"
  >
    <div
      aria-hidden="true"
      className="k-ToggleSwitch__circle"
    />
  </button>
  <label
    className="sc-htpNat dbfzSp"
    htmlFor="toggle-switch"
    id="toggle-switch_label"
  >
    switch
  </label>
</div>
`;
